cmds/core/sluinit/uinit.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"github.com/u-root/u-root/pkg/cmdline"
+	slaunch "github.com/u-root/u-root/pkg/securelaunch"
+	"github.com/u-root/u-root/pkg/securelaunch/policy"
+	"github.com/u-root/u-root/pkg/securelaunch/tpm"
+	"log"
+	"os"
+)
+
+func main() {
+
+	slaunch.Debug("********Step 1: init completed. starting main ********")
+	tpmDev, err := tpm.GetHandle()
+	if err != nil {
+		log.Printf("tpm.getHandle failed. err=%v", err)
+		os.Exit(1)
+	}
+	defer tpmDev.Close()
+
+	slaunch.Debug("********Step 2: locate SL Policy ********")
+	rawBytes, err := policy.Locate()
```

Comment:
i feel like the policy library should just parse the policy internally and return it here.

if you don't think so, i'm curious what can a user of the policy library do with a policy other than parse it

Reply:
Can u please elab on this ...not getting u.. Are u saying no need to do separate fns Locate and Parse ?
_OR_ you dont want these fns in a separate package ?
As per our recent discussion, after moving the scanBlockDevice and scanKernelCmdLine inside policy package, things should be pretty organized no ?

Reply:
i was thinking that locate and parse should be internal functions to policy, and you just have one

```go
func Policy() (*Policy, error) {
  b, err := locate()
  if err != nil {
     return nil, err
  }
  policy, err := parse(b)
  if err != nil {
     return nil, err
  }
  if policy == nil {
     return nil, fmt.Errorf("no policy found")
  }
  return policy, nil
}
```

Reply:
I can't see why a user of the policy package would want the raw bytes of the policy -- I think a user of the policy package would just want the *Policy object

and everything below that should be internal -- locate, parse, scanBlockDev, scanKernelCmdline, etc

wdyt?